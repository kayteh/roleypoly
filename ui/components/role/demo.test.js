/* eslint-env jest */
import * as React from 'react'
import { shallow } from 'enzyme'
import RoleDemo from './demo'

describe('<RoleDemo />', () => {
  it('renders', () => {
    const demo = shallow(<RoleDemo role={{ name: 'test demo role', color: '#ffffff' }} />)
    expect(demo.html()).toMatchSnapshot()
  })

  it('changes state when clicked', () => {
    const demo = shallow(<RoleDemo role={{ name: 'test demo role', color: '#ffffff' }} />)
    expect(demo.state().active).toEqual(false)
    demo.dive().simulate('click')
    expect(demo.state().active).toEqual(true)
  })
})
