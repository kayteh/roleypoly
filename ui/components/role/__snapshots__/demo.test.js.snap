// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<RoleDemo /> renders 1`] = `"<div style=\\"--role-color-outline:hsla(0, 0%, 100%, 0.7);--role-color-outline-alt:hsla(0, 0%, 100%, 0.4);--role-color-active:hsl(0, 0%, 100%);--role-color-base:hsl(0, 0%, 100%)\\" class=\\"rolestyled-r4hjov-0 kfzQxT\\">test demo role</div>"`;
